new-bot/src/bot/helpers/download-file.ts

```
@@ -0,0 +1,55 @@
+      response.data.pipe(writer);
+      writer.on('finish', () => resolve(filePath));
+      writer.on('error', reject);
+      setTimeout(() => {
+        writer.close();
+        reject(new Error(`Download timed out after ${timeout}ms`));
+      }, timeout);
+    } catch (error) {
+      console.error("Error downloading file:", error);
+      throw error;
+    }
+  })
```

Biome

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)